core/src/edu/cornell/gdiac/temporary/LevelSelect.java

```diff
@@ -177,51 +180,63 @@ public void populate(AssetDirectory assetDirectory) {
-//        allLevels= new String[numSongs *3];
-        albumCoverCoords = new Vector2[3];
-//        System.out.println("num levels "+numLevels);
+        albumCoverCoords = new Vector2[allLevels.length];
+        albumScales = new float[allLevels.length];
         gameplayController = new GameplayController(canvas.getWidth(),canvas.getHeight());
         albumCovers = new Texture[numSongs];
-
         playButton = directory.getEntry("ghost-play",Texture.class);
         easyButton = directory.getEntry("easy",Texture.class);
         mediumButton = directory.getEntry("medium",Texture.class);
         hardButton = directory.getEntry("hard",Texture.class);
-
         goLeft = directory.getEntry("level-select-left",Texture.class);
         goRight = directory.getEntry("level-select-right",Texture.class);
-
-        goLeftCoords = new Vector2(goLeft.getWidth(),canvas.getHeight()/2);
-        goRightCoords = new Vector2(canvas.getWidth()-goLeft.getWidth(),canvas.getHeight()/2);
-
-
-        levelBackground = new FilmStrip(directory.getEntry("level-select-background", Texture.class), 1, 1);
-
-        playButtonCoords = new Vector2(canvas.getWidth()/2, canvas.getHeight()/8);
-        mediumButtonCoords = new Vector2(canvas.getWidth()/2 , canvas.getHeight()/4);
-        easyButtonCoords = new Vector2(mediumButtonCoords.x-easyButton.getWidth(), canvas.getHeight()/4);
-        hardButtonCoords = new Vector2(mediumButtonCoords.x+easyButton.getWidth(), canvas.getHeight()/4);
-
+        levelBackground = directory.getEntry("level-select-background", Texture.class);
 //      album covers are called 1, 2, 3 and so on in assets.json
         for (int i = 0; i < numSongs; i++){
             albumCovers[i] = directory.getEntry(Integer.toString(i+1),Texture.class);
         }
 
-        albumCoverCoords[0]=new Vector2((canvas.getWidth()/4-10),(canvas.getHeight()*2/3)-25);
-        albumCoverCoords[1]=new Vector2((canvas.getWidth()/2 ),(canvas.getHeight()*2/3)-25);
-        albumCoverCoords[2]=new Vector2((canvas.getWidth()/2 )+ (canvas.getWidth()/4)+10,(canvas.getHeight()*2/3)-25);
+        setCoords(canvas.getWidth(),canvas.getHeight());
+    }
+
+    /**
+     * loadCoords set the coordinates of all assets
+     */
+    public void setCoords(int width, int height) {
+        float sx = ((float)width)/STANDARD_WIDTH;
+        float sy = ((float)height)/STANDARD_HEIGHT;
+
+        scale = (sx < sy ? sx : sy);
+        centerScale = 0.65f*scale;
+        cornerScale = 0.45f*scale;
+        goBackCoords=new Vector2 (width/12f, height*9f/10f);
+        goLeftCoords = new Vector2(width/10f,height/2f);
+        goRightCoords = new Vector2(width-(width/10f),height/2f);
+        playButtonCoords = new Vector2(width/2f, height/8f);
+        mediumButtonCoords = new Vector2(width/2f , height/4f);
+        easyButtonCoords = new Vector2((width/2f)-(width/10f), height/4f);
+        hardButtonCoords = new Vector2((width/2f)+(width/10f), height/4f);
```

Comment:
do not initialize Vector2 coordinates anymore, and this is even worse since I think every time you resize the screen, it initializes new coordinates.